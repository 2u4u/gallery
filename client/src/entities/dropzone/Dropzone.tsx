import { useCallback } from 'react';
import { useDropzone } from 'react-dropzone'
import { PreviewType } from 'shared/types/index.type';
import { getFileType } from 'shared/utils/helpers';
import './index.scss';

type Props = {
    setPreviews: (preview: PreviewType[] | ((prevPreview: PreviewType[]) => PreviewType[])) => void;
}

function Dropzone({ setPreviews }: Props) {
    const handleFileDrop = useCallback(async (acceptedFiles) => {
        const createPreview = (file: File) => {
            const type = getFileType(file.type);
            let preview: PreviewType;
            if (type === 'image') {
                preview = {
                    type: 'image',
                    url: URL.createObjectURL(file),
                    name: file.name,
                    file: file,
                }
            } else {
                preview = {
                    type,
                    url: `./img/${type}.svg`,
                    name: file.name,
                    file: file,
                };
            }
            setPreviews((previews: PreviewType[]) => [...previews, preview]);
        }

        acceptedFiles.forEach((file: File) => createPreview(file))
    }, [setPreviews]);

    const {getRootProps, getInputProps, isDragActive} = useDropzone({ 
        onDrop: handleFileDrop, 
        accept: '.jpeg, .png, .mpeg, .wav, .mp4, .webm, .mp3', 
    });

    return (
        <div {...getRootProps()} className='dropzone'>
            <input {...getInputProps()} />
            {
                isDragActive ?
                <p>Drop the files here ...</p> :
                <p>Drag and drop some files here, or click to select files</p>
            }
        </div>
    );
}

export default Dropzone;
