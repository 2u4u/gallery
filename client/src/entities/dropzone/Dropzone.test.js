import renderer from 'react-test-renderer';
import Dropzone from './Dropzone';

describe('Dropzone component ', () => {
    it('matches snapshot', () => {
        const setPreviews = () => false;
        const component = renderer.create(
            <Dropzone 
                setPreviews={setPreviews}
            />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
