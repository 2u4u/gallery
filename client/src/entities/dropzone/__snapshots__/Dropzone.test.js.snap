// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone component  matches snapshot 1`] = `
<div
  className="dropzone"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <input
    accept=".jpeg, .png, .mpeg, .wav, .mp4, .webm, .mp3"
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <p>
    Drag and drop some files here, or click to select files
  </p>
</div>
`;
